### ISSUE TITLE
Dockerfile builds an image that will not start

### ISSUE THREAD

**Issue by Lnk2past** [clulab/eidos#675]
Hello! I am a newbie with Eidos and all, but I ran into an issue when running the container built from the provided Dockerfile.

The web service fails to start with the following error (and it repeats many, many times over)
```
[ERROR] [09/25/2019 14:30:53.685] [play-dev-mode-akka.actor.default-dispatcher-9] [akka.actor.ActorSystemImpl(play-dev-mode)] Internal server error, sending 500 response
Configuration error: Configuration error[application.conf @ file:/eidos/webapp/target/scala-2.12/classes/application.conf: 16: Key '-e' may not be followed by token: 'play.filters.hosts' (if you intended 'play.filters.hosts' to be part of a key or string value, try enclosing the key or value in double quotes)]
        at play.api.Configuration$.configError(Configuration.scala:155)
        at play.api.Configuration$.load(Configuration.scala:101)
        at play.api.Configuration$.load(Configuration.scala:109)
        at play.api.Configuration$.load(Configuration.scala:115)
        at play.api.http.DefaultHttpErrorHandler$.setEditor$lzycompute(HttpErrorHandler.scala:277)
        at play.api.http.DefaultHttpErrorHandler$.setEditor(HttpErrorHandler.scala:276)
        at play.api.http.DefaultHttpErrorHandler$.onServerError(HttpErrorHandler.scala:285)
        at play.core.server.Server.logExceptionAndGetResult$1(Server.scala:52)
        at play.core.server.Server.getHandlerFor(Server.scala:82)
        at play.core.server.Server.getHandlerFor$(Server.scala:48)
        at play.core.server.AkkaHttpServer.getHandlerFor(AkkaHttpServer.scala:41)
        at play.core.server.AkkaHttpServer.getHandler(AkkaHttpServer.scala:214)
        at play.core.server.AkkaHttpServer.handleRequest(AkkaHttpServer.scala:194)
        at play.core.server.AkkaHttpServer.$anonfun$createServerBinding$3(AkkaHttpServer.scala:106)
        at akka.stream.impl.fusing.MapAsync$$anon$23.onPush(Ops.scala:1172)
        at akka.stream.impl.fusing.GraphInterpreter.processPush(GraphInterpreter.scala:499)
        at akka.stream.impl.fusing.GraphInterpreter.processEvent(GraphInterpreter.scala:462)
        at akka.stream.impl.fusing.GraphInterpreter.execute(GraphInterpreter.scala:368)
        at akka.stream.impl.fusing.GraphInterpreterShell.runBatch(ActorGraphInterpreter.scala:571)
        at akka.stream.impl.fusing.GraphInterpreterShell$AsyncInput.execute(ActorGraphInterpreter.scala:457)
        at akka.stream.impl.fusing.GraphInterpreterShell.processEvent(ActorGraphInterpreter.scala:546)
        at akka.stream.impl.fusing.ActorGraphInterpreter.akka$stream$impl$fusing$ActorGraphInterpreter$$processEvent(ActorGraphInterpreter.scala:725)
        at akka.stream.impl.fusing.ActorGraphInterpreter$$anonfun$receive$1.applyOrElse(ActorGraphInterpreter.scala:740)
        at akka.actor.Actor.aroundReceive(Actor.scala:514)
        at akka.actor.Actor.aroundReceive$(Actor.scala:512)
        at akka.stream.impl.fusing.ActorGraphInterpreter.aroundReceive(ActorGraphInterpreter.scala:650)
        at akka.actor.ActorCell.receiveMessage(ActorCell.scala:527)
        at akka.actor.ActorCell.invoke(ActorCell.scala:496)
        at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:257)
        at akka.dispatch.Mailbox.run(Mailbox.scala:224)
        at akka.dispatch.Mailbox.exec(Mailbox.scala:234)
        at akka.dispatch.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
        at akka.dispatch.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
        at akka.dispatch.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
        at akka.dispatch.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
Caused by: com.typesafe.config.ConfigException$Parse: application.conf @ file:/eidos/webapp/target/scala-2.12/classes/application.conf: 16: Key '-e' may not be followed by token: 'play.filters.hosts' (if you intended 'play.filters.hosts' to be part of a key or string value, try enclosing the key or value in double quotes)
        at com.typesafe.config.impl.ConfigDocumentParser$ParseContext.parseError(ConfigDocumentParser.java:201)
        at com.typesafe.config.impl.ConfigDocumentParser$ParseContext.parseError(ConfigDocumentParser.java:197)
        at com.typesafe.config.impl.ConfigDocumentParser$ParseContext.parseObject(ConfigDocumentParser.java:459)
        at com.typesafe.config.impl.ConfigDocumentParser$ParseContext.parse(ConfigDocumentParser.java:646)
        at com.typesafe.config.impl.ConfigDocumentParser.parse(ConfigDocumentParser.java:14)
        at com.typesafe.config.impl.Parseable.rawParseValue(Parseable.java:260)
        at com.typesafe.config.impl.Parseable.rawParseValue(Parseable.java:248)
        at com.typesafe.config.impl.Parseable.parseValue(Parseable.java:180)
        at com.typesafe.config.impl.Parseable.parseValue(Parseable.java:174)
        at com.typesafe.config.impl.Parseable.parseValue(Parseable.java:307)
        at com.typesafe.config.impl.Parseable$ParseableResources.rawParseValue(Parseable.java:747)
        at com.typesafe.config.impl.Parseable$ParseableResources.rawParseValue(Parseable.java:710)
        at com.typesafe.config.impl.Parseable.parseValue(Parseable.java:180)
        at com.typesafe.config.impl.Parseable.parseValue(Parseable.java:174)
        at com.typesafe.config.impl.Parseable.parse(Parseable.java:152)
        at com.typesafe.config.impl.SimpleIncluder.fromBasename(SimpleIncluder.java:185)
        at com.typesafe.config.impl.ConfigImpl.parseResourcesAnySyntax(ConfigImpl.java:132)
        at com.typesafe.config.ConfigFactory.parseResourcesAnySyntax(ConfigFactory.java:978)
        at com.typesafe.config.DefaultConfigLoadingStrategy.parseApplicationConfig(DefaultConfigLoadingStrategy.java:36)
        at com.typesafe.config.ConfigFactory.defaultApplication(ConfigFactory.java:473)
        at play.api.Configuration$.$anonfun$load$5(Configuration.scala:73)
        at scala.Option.getOrElse(Option.scala:121)
        at play.api.Configuration$.load(Configuration.scala:69)
        ... 33 more
```

I came across #666 and figured I would try removing the line that was added in #667. Removing the line added in that PR allowed the container to startup.

I am running on Windows 10 with Docker 19.03.2, build 6a30dfc. I can submit a PR back, but wanted to confirm the issue I am seeing.

**Comment by bgyori** [comment]
Sorry if I messed something up, we need to double check that #667 adds that line correctly to each of the Dockerfiles, it works for me though.

**Comment by kwalcock** [comment]
Thanks for the report and I'm sorry your initial experience was disappointing.  It appears that the -e option in that line doesn't work on your platform.
```
RUN echo -e "\nplay.filters.hosts {allowed = [\".:9000\"]}" >> ./webapp/conf/application.conf
```
Although Eidos runs directly on Windows, I've never tried to build the Docker file on Windows and was under the impression that it is not a straightforward task.  How are you managing to do that?  Is the echo being used the Windows version?  That might explain it not understanding -e.

**Comment by BeckySharp** [comment]
Is the built docker image available on Dockerhub? and if so/it were would
that "solve" this?

**Comment by Lnk2past** [comment]
@kwalcock I wouldn't use the word disappointing! I was able to get things running directly on Windows perfectly fine the other day, and was able to get everything strung together with INDRA and a Jupyter notebook. As far as building the images and running, I simply followed the instructions as they are for building the image and then running the containers. I waited a few minutes for everything to initialize and once it was ready it started working.

@bgyori I do not really know why the echo statements would be treated differently, and honestly I did not even realize that that is what is happening until you both responded. I am going to fudge it a bit and see if I can get it working with that line.

@BeckySharp Having the image available would fix it for those that wish to pull it down for sure, but for anyone building the image having a fix in place within the Dockerfile itself is much more preferable (at least in my opinion it is).

**Comment by Lnk2past** [comment]
I changed the line to this:

```
RUN ["/bin/bash", "-c", "echo -e '\nplay.filters.hosts {allowed = [\".:9000\"]}' >> ./webapp/conf/application.conf"]
```

Everything seems to be working fine in my environment. If anyone with a Mac or Linux environment can confirm that this works there as well that would be awesome.

**Comment by Lnk2past** [comment]
Sorry for the post-spam, but is there a way to more effectively communicate with everyone who knows how to use and debug Eidos (slack, discord, etc.)? I do not want to open up issues all the time when I have a question =)

**Comment by kwalcock** [comment]
Knock on the wall and I will turn on Slack?

**Comment by kwalcock** [comment]
I'm not sure of the best way, but one way is to keep adding to this issue, which apparently triggers quite a few notifications.  You might also start an issue, like "@Lnk2past has a question".  That might help it get routed to the people who might be able to answer them.

Thanks for the email which arrived midway through this response and included more of your credentials.

**Comment by kwalcock** [comment]
I'm not sure that "/bin/bash" is the best way.  In an example at https://docs.docker.com/engine/reference/builder/ they just use "sh", so maybe "bash" in case it is located elsewhere.  I'm running a test on Linux and Mac.

**Comment by Lnk2past** [comment]
No worries. Thanks for testing it out and such. Just saw the PR you opened 👍

**Comment by kwalcock** [comment]
Hopefully solved with #680.
